server/server_test.go

```diff
@@ -47,4 +51,27 @@ var _ = Describe("Server", func() {
 		})
 	})
 
+	Describe("POST the /signup endpoint", func() {
+		BeforeEach(func() {
+			response = performRequest(router, "POST", "/api/v1/signup", jsonSignUp)
+		})
+		It("Returns with Status 201", func() {
+			Expect(response.Code).To(Equal(201))
+		})
+		It("Added a new User", func() {
+			newuser := models.User{}
+			json.Unmarshal([]byte(jsonSignUp), &newuser)
+			Expect(data.Users().List[len(data.Users().List)-1]).To(Equal(newuser))
```

Comment:
A better way to test would be call the function to read from database. And check if the unmarshalled data from database same as json data passed to the POST userRegistration

Reply:
yes! that makes sense!